src/components/textarea/textarea.tsx

```diff
@@ -0,0 +1,321 @@
+import {
+  Component,
+  h,
+  Prop,
+  Element,
+  Event,
+  EventEmitter,
+  VNode,
+  Watch,
+  Method,
+  Host,
+  State
+} from "@stencil/core";
+import { connectForm, disconnectForm, FormComponent, HiddenFormInputSlot } from "../../utils/form";
+import { connectLabel, disconnectLabel, getLabelText, LabelableComponent } from "../../utils/label";
+import { getSlotted } from "../../utils/dom";
+import { CSS, ERROR_MESSAGES, SLOTS } from "./resources";
+import {
+  connectLocalized,
+  disconnectLocalized,
+  LocalizedComponent,
+  NumberingSystem,
+  numberStringFormatter
+} from "../../utils/locale";
+import { createObserver } from "../../utils/observers";
+@Component({
+  tag: "calcite-textarea",
+  styleUrl: "textarea.scss",
+  shadow: true
+})
+export class Textarea implements FormComponent, LabelableComponent, LocalizedComponent {
+  //--------------------------------------------------------------------------
+  //
+  //  Element
+  //
+  //--------------------------------------------------------------------------
+  @Element() el: HTMLCalciteTextareaElement;
+
+  //--------------------------------------------------------------------------
+  //
+  //  Properties
+  //
+  //--------------------------------------------------------------------------
+
+  /** When `true`, focuses the `textarea` element on page render. */
+  @Prop({ reflect: true }) autofocus: boolean;
+
+  /** When `true`, disables the component. */
+  @Prop({ reflect: true }) disabled: boolean;
+
+  /** Specifies the placeholder text for the input. */
+  @Prop({ reflect: true }) placeholder: string;
+
+  /** Whne `true`, the component's value can be read, but cannot be modified.  */
+  @Prop({ reflect: true }) readonly: boolean;
+
+  /** Specifies number or rows allowed. */
+  @Prop({ reflect: true }) rows: number;
+
+  /** Specifies number or columns allowed. */
+  @Prop({ reflect: true }) cols: number;
+
+  /** Specifies maximum number of characters allowed. */
+  @Prop({ reflect: true }) maxlength: number;
+
+  /** Specifies name of the component  */
+  @Prop({ reflect: true }) name: string;
+
+  /** Specifies the size of `textarea` component. */
+  @Prop({ reflect: true }) scale: "l" | "m" | "s" = "m";
+
+  /** Specifies wrapping mechanism for the text.  */
+  @Prop({ reflect: true }) wrap: "soft" | "hard" = "soft";
+
+  // @Prop() autocomplete: boolean;
+
+  /** The component's value. */
+  @Prop({ mutable: true }) value: string;
+
+  /** When `true` , footer will be added to the component. */
+  @Prop({ reflect: true }) footer = false;
+
+  /** When `true`, disables the resizing handle. */
+  @Prop({ reflect: true }) resizeDisabled: boolean;
```

Comment:
@benelan @geospatialem @alisonailea @eriklharper It seems like our components have `disableFeature` and `featureDisabled` as props throughout our components.

For example, combobox has `filterDisabled` whereas dropdown has `disableCloseOnSelect`. I think we need to choose a format and update all of our components. 

Was this brought up in consistency issues before?

Reply:
I think I prefer the way @anveshmekala has it as it shows as `resize-disabled` which is consistent with `aria-disabled="true"`

Reply:
FWIW, I'd say that the pattern `<property>Disabled` makes sense if the type is `boolean` and the pattern `disable<property-or-feature>` makes sense if this is a function that performs the disabling.  Since the vast majority of these types of props are likely `boolean` my guess is we'll want the former rather than the latter.

Reply:
I added it to this issue: https://github.com/Esri/calcite-components/issues/4688

Reply:
To add more context, the combobox `filterDisabled` prop needed to be renamed and was added to [API consistency list](https://github.com/Esri/calcite-components/wiki/Consistency-Audit) (item 12)  and I agree with using <property>Disabled to indicate booleans.